### ISSUE TITLE
Enable optional Owned Objects

### ISSUE THREAD

**Issue by frbatist** [dotnet/efcore#10818]
There's no support for defining a Owned Object as optional (AKA Nullable).

We have classes like "User" or "Address", that are ValueObjects, not actual entities.
In some cases we want this "User" to be required, and in other cases to be optional.

It would be nice to be able to map as optional, so migrations create the columns as not null.

Something like this:

```C#
orderConfiguration.OwnsOne(p => p.OrderDetails, cb =>
    {
        cb.OwnsOne(c => c.BillingAddress).IsRequired(false);
        cb.OwnsOne(c => c.ShippingAddress);
    });
```

**Comment by AndriySvyryd** [comment]
To be useful for relational providers this requires table splitting support https://github.com/aspnet/EntityFrameworkCore/issues/9005

**Comment by ajcvickers** [comment]
Closing this as a duplicate of #9005 since this is the real feature that needs implementing for optional owned entities. API for configuration will be added as part of implementing that feature.

**Comment by cetk** [comment]
Is it resolved in EF core 2.2 , I am getting same error ?

**Comment by ajcvickers** [comment]
@cetk This issue is closed as a duplicate. Therefore, any information will be contained in the referenced issue--that is #9005--which is closed as fixed in milestone "3.0.0-preview4". This means it is fixed for 3.0, not 2.2.
